zanata-war/src/main/java/org/zanata/events/DocumentLocaleKey.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.events;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.zanata.common.LocaleId;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+
+/**
+ * @author Alex Eng<a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@AllArgsConstructor
+@EqualsAndHashCode
+@Getter
+public class DocumentLocaleKey implements Serializable{
+
+    // this may be null in the case of document uploads
+    private final @Nullable Long actorId;
```

Comment:
This field doesn't fit with the name of the class, but rather than rename the class, would it be possible to factor it out of here? A single TextFlowTargetStateEvent should only be for a single actor.